File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.swift
func solution(
    _ denum1:Int,
    _ num1:Int,
    _ denum2:Int,
    _ num2:Int
) -> [Int] {
    // 분모를 공통으로 만들기 위해 각 분자에 다른 분모 값을 곱해줌
    let up = (denum1 * num2) + (denum2 * num1) // 분자끼리 더해준 값
    let down = num1 * num2 // 공통 분모 값
    var result = [up, down]
    
    // 기약분수로 만드는 단계 
    for i in (1...down) {
        if (up % i == 0) && (down % i == 0) {
            result[0] = up / i
            result[1] = down / i
        }
    }
    return result
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.py
import sys
input = sys.stdin.readline

def get_gcd(num1, num2):
    return get_gcd(num2, num1 % num2) if num1 % num2 else num2

def solution(numer1, denom1, numer2, denom2):
    answer = []
    num = (numer1 * denom2) + (denom1 * numer2)
    dem = denom1 * denom2
    min = get_gcd(num, dem)
    print(min)
    answer.append(num//min)
    answer.append(dem//min)

    return answer